mappings/net/minecraft/item/map/MapState.mapping

```diff
@@ -1,19 +1,69 @@
 CLASS cnv net/minecraft/item/map/MapState
+	CLASS cnv$a PlayerUpdateTracker
+		FIELD a player Lavx;
+		FIELD d dirty Z
+		FIELD e startX I
+		FIELD f startZ I
+		FIELD g endX I
+		FIELD h endZ I
+		FIELD i emptyPacketsRequested I
+		METHOD <init> (Lcnv;Lavx;)V
+			ARG 1 player
+		METHOD a markDirty (II)V
+			ARG 1 x
+			ARG 2 z
+		METHOD a getPacket (Lbca;)Lka;
 	FIELD a xCenter I
 	FIELD b zCenter I
 	FIELD c dimension Lbyd;
 	FIELD d showIcons Z
 	FIELD e unlimitedTracking Z
 	FIELD f scale B
-	FIELD g colorArray [B
+	FIELD g colors [B
 	FIELD h locked Z
+	FIELD i updateTrackers Ljava/util/List;
 	FIELD j icons Ljava/util/Map;
+	FIELD k updateTrackersByPlayer Ljava/util/Map;
+	FIELD l banners Ljava/util/Map;
+	FIELD m frames Ljava/util/Map;
 	METHOD <init> (Ljava/lang/String;)V
 		ARG 1 key
 	METHOD a calculateCenter (DDI)V
+	METHOD a markDirty (II)V
+		ARG 1 x
+		ARG 2 z
 	METHOD a init (IIIZZLbyd;)V
 		ARG 3 scale
 		ARG 4 showIcons
 		ARG 5 unlimitedTracking
+	METHOD a getPlayerSyncData (Lavx;)Lcnv$a;
+		ARG 1 player
+	METHOD a update (Lavx;Lbca;)V
+		ARG 1 player
+		ARG 2 stack
+	METHOD a getPlayerMarkerPacket (Lbca;Lbgr;Lavx;)Lka;
+		ARG 1 map
+		ARG 2 world
+		ARG 3 pos
 	METHOD a addDecorationsTag (Lbca;Lev;Ljava/lang/String;Lcns$a;)V
 		ARG 2 id
+	METHOD a removeBanner (Lbgr;II)V
+		ARG 1 world
+		ARG 2 x
+		ARG 3 z
+	METHOD a addBanner (Lbhi;Lev;)V
+		ARG 1 world
+		ARG 2 pos
+	METHOD a addIcon (Lcns$a;Lbhi;Ljava/lang/String;DDDLjm;)V
```

Comment:
addDecoration? Decoration is how Mojang calls exploration map markers, see e.g. the `exploration_map` loot table function.

Reply:
The NBT name is "icon"...

Reply:
https://minecraft.gamepedia.com/Player.dat_format#Maps
It's stored in `Decorations`...

Reply:
Which NBT?

Reply:
I think the format changed in 1.14. I can't check right now, but the Minecraft wiki link seems very different from what I saw.

Reply:
support rune. wiki is always outdated. i even added many stuff to wiki (e.g. how hero of the village discounts, raids.dat format) based on my observations in yarn.

Reply:
Nope, didn't change: https://imgur.com/a/FVgEQ6F

Reply:
"Decorations" are the icons that are not banner or item frame icons. Decorations, banners, and frames are all called "icons", as shown in the map update packet:

```java
    public void apply(MapState mapState) {
        mapState.scale = scale;
        mapState.showIcons = showIcons;
        mapState.locked = locked;
        mapState.icons.clear();
        for (int i = 0; i < icons.length; ++i) {
            MapIcon icon = icons[i];
            mapState.icons.put("icon-" + i, icon);
        }

        for (int x = 0; x < width; ++x) {
            for (int z = 0; z < height; ++z) {
                mapState.colors[startX + x + (startZ + z) * 128] = colors[x + z * width];
            }
        }
    }
```

Reply:
Ah, it is both things, than it makes sense.
Yes, decorations are stored in item nbt, while banners and map markers are stored in the map data itself.